From 5b0b07533a7646817a66aea8c7aaa9ab84c9d050 Mon Sep 17 00:00:00 2001
From: Catalin Popescu <catalin.popescu@intel.com>
Date: Tue, 8 Sep 2015 13:52:35 +0200
Subject: [PATCH] DC TI PMIC: disable power button support.

Power button is already supported by gpio-keys driver which delivers
input events to Android. Any redundancy in input events delivered to
user space will end up with strange behaviors such as screen toggling
on and off almost immediately.

Change-Id: I98e670d94fa146273e434f4183b87323e219f7f6
Signed-off-by: Catalin Popescu <catalin.popescu@intel.com>
---
 drivers/mfd/intel_soc_pmic_dc_ti.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/mfd/intel_soc_pmic_dc_ti.c b/drivers/mfd/intel_soc_pmic_dc_ti.c
index 827032988eec..7ccaca60ab81 100644
--- a/drivers/mfd/intel_soc_pmic_dc_ti.c
+++ b/drivers/mfd/intel_soc_pmic_dc_ti.c
@@ -38,6 +38,10 @@ enum {
 	CCEOCAL = 7,
 };
 
+/*If need PMIC support power button event report, define it*/
+/*#define PMIC_DC_TI_PWR_BTN_EN*/
+
+#ifdef PMIC_DC_TI_PWR_BTN_EN
 static struct resource power_button_resources[] = {
 	{
 		.start	= PWRBTN,
@@ -45,6 +49,7 @@ static struct resource power_button_resources[] = {
 		.flags	= IORESOURCE_IRQ,
 	},
 };
+#endif
 
 static struct resource thermal_resources[] = {
 	{
@@ -90,12 +95,14 @@ static struct mfd_cell dollar_cove_ti_dev[] = {
 		.num_resources = ARRAY_SIZE(adc_resources),
 		.resources = adc_resources,
 	},
+#ifdef PMIC_DC_TI_PWR_BTN_EN
 	{
 		.name = "dollar_cove_ti_power_button",
 		.id = 0,
 		.num_resources = ARRAY_SIZE(power_button_resources),
 		.resources = power_button_resources,
 	},
+#endif
 	{
 		.name = "dollar_cove_ti_pwrsrc",
 		.id = 0,
-- 
2.6.0.rc0.2.g7662973

